### PR TITLE
[Annotation plugin] Add instrumentation test for annotation update

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[Annotation plugin] Add instrumentation test for annotation update</changelog>`.

### Summary of changes

This pr add an instrumentation test which will update the position of annotation while changing styles to verify no crash happens during this process.
QQ20210201-165504

### CODE REVIEW

File: app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    rule.scenario.onActivity {
+      val handler = Handler(it.mainLooper)
+
+      it.runOnUiThread {
+        mapboxMap.loadStyleUri(AnnotationUtils.STYLES[index++ % AnnotationUtils.STYLES.size]) {
+          symbolManager = mapView.getAnnotationPlugin().getSymbolManager(mapView)
+          symbol = symbolManager.create(
+            SymbolOptions()
+              .withIconColor(ColorUtils.colorToRgbaString(Color.RED))
```
Comment (kiryldz) [inline app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:49]: can we add maximum possible options here?

Comment (Chaoba) [reply app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:49]: The purpose of this test is to verify no crash happen while changing style and update at the same time. That's said, the inside source and layer are handled properly, which has been done in https://github.com/mapbox/mapbox-maps-android/pull/47/files#diff-4439957e75cc10a9ecf0041db71889e34906978146ed8b09519c8e0cc7405a1fR142. No matter how many options are added here actually has no effect for it.

Comment (kiryldz) [reply app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:49]: But if some option could not get applied and causes a crash - we could catch it here, couldn't we?
Leaving the decision up to you here.

**Review by kiryldz — APPROVED** [review]

File: app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt (hunk 1 above)
Comment (Chaoba) [reply app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:49]: 🤔 , Let's add some more options to cover a wider scope.

**Pull request merged**